File: PROGRAMMERS/콜라츠추측.py
# n = 6
# n = 16
n = 626331

def solution(n):
    answer = 0
    while True:
        if n == 1:
            break
        elif n % 2 == 0:
            answer += 1
            n = n // 2
        elif n % 2 != 0:
            answer += 1
            n = (n * 3) + 1
        if answer >= 500:
            return -1
    return answer

print(solution(n))

File: PROGRAMMERS/JAVA/src/Lv_1/콜라츠추측.java
package Lv_1;

public class 콜라츠추측 {
    public int solution(int num) {
        int answer = -1;
        int input_num = num;
        int cnt = 1;
        if (num == 1) {
            return 0;
        }
        for (int i = 0; i < 500; i++) {
            if (input_num % 2 == 0) {
                input_num = input_num / 2;
            } else if (input_num % 2 == 1) {
                input_num = (input_num * 3) + 1;
            }
            if (input_num == 1) {
                answer = cnt;
                break;
            }
            cnt++;
        }
        return answer;
    }
}
